### DISCUSSION TITLE
Modifying ActivityPub posts via Wordpress filter?

### DISCUSSION THREAD

**Discussion by craigconstantine** [Automattic/wordpress-activitypub#228] [Q&A] [answered]
I noticed that `to_array()` in https://github.com/pfefferle/wordpress-activitypub/blob/44c652eba8174368c9fdc2ca33e02fd2afbe436d/includes/model/class-post.php#L43 provides for filters for `activitypub_post`. But when I tried to hook the following, it didn't have an effect.

So my first question is: Is this the right place to try hooking?

I understand these are some ugly preg_replace()'s. It's just copy-n-paste from another filter I've tuned for use on my RSS feed posts.

```
// custom munging of posts going out via ActivityPub
add_filter( 'activitypub_post', 'craig_activitypub_post_filter' );
function craig_activitypub_post_filter( $post /* an ActivityPub array, not a WP Post object */ ) {

	// inline minimal styling for blockquotes
	$post['content'] = preg_replace(
		'/<blockquote/i',
		'<blockquote style="margin-left: 20px; border-left: 2px solid #f2777a; padding-left: 10px;" ',
		$post['content']
	);

	// remove slip address gutenburg blocks
	$post['content'] = preg_replace(
		'/<div class="wp-block-gutenberg-sladdr-block-address slip-address">\s*<p>[^<]*<\/p>\s*<\/div>/i',
		'',
		$post['content']
	);

	// WP generates CITE tags for the citation portion of a blockquote. This doesn't
	// work well with email clients and RSS readers and such. We turn them back
	// in P tags, add italics and try some opacity via embedded styling.
	$post['content'] = preg_replace( '/<cite>/i', '<p><i>', $post['content'] );
	$post['content'] = preg_replace( '/<\/cite>/i', '</i></p>', $post['content'] );

	return $post;
}

```

**Comment by pfefferle** [comment, marked as answer]
It is/should be the right place (let me know if you need some more Parameters for the filter!). To check if your changes are working you can add `/activitypub` at the end of each post, to see the result. Otherwise it is hard to check, because mastodon and others might strip html elements. 

Btw. Here is an example https://notiz.blog/2022/11/14/the-at-protocol/activitypub/

**Reply by craigconstantine** [reply]
Terrific. The `/activitypub` tip made it easy to troubleshoot. Thanks.
